curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a23d1c5f1c93161f3582ae.md

```diff
@@ -0,0 +1,104 @@
+---
+id: 62a23d1c5f1c93161f3582ae
+title: Step 6
+challengeType: 0
+dashedName: step-6
+---
+
+# --description--
+
+Similar to your `#stats` element, your `#monsterStats` element needs two `span` elements. Give them the class `stat` and give the first element the text `Monster Name: ` and the second the text `Health: `. After the text in each, add a `strong` element with an empty nested `span` element.
+
+# --hints--
+
+Your `monsterStats` element should have two `span` elements.
+
+```js
+const spans = document.querySelectorAll(`#monsterStats > span`);
+assert.equal(spans.length, 2);
+```
+
+Your new `span` elements should both have a `class` value of `stat`.
+
+```js
+const spans = document.querySelectorAll(`#monsterStats > span`);
+assert.equal(spans[0].className, 'stat');
+assert.equal(spans[1].className, 'stat');
+```
+
+Your first `span` element should have the text `Monster Name: `. Make sure the spacing is correct.
+
+```js
+const spans = document.querySelectorAll(`#monsterStats > span`);
+assert.equal(spans[0].innerText, 'Monster Name: ');
+```
+
+Your second `span` element should have the text `Health: `. Make sure the spacing is correct.
+
+```js
+const spans = document.querySelectorAll(`#monsterStats > span`);
+assert.equal(spans[1].innerText, 'Health: ');
```

Comment:
I don't know what is happening, but you are getting `Health:` here instead of `Health: `

Reply:
Changing this to `textContent` to avoid automatically trimming the blank space seems to fix it?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent, It seems it will get `<strong>` and `<span>` too, but it will work. 

I think it's better to change the test to check for `Health:` because it's JavaScript steps, so we can go lenient.

We can go regex too, but I prefer nothing over the regex.